src/pkg/retention/policy/rule/index/index.go

```diff
@@ -48,13 +63,78 @@ type IndexedParam struct {
 
 // indexedItem is the item saved in the sync map
 type indexedItem struct {
-	Meta *IndexMeta
+	Meta *Metadata
+
+	Factory rule.Factory
+}
 
-	Factory Factory
+func init() {
+	// Register latest pushed
+	Register(&Metadata{
```

Comment:
Ditto my comments on other `Register`-like calls in `init` functions. It feels weird to me to call these from a different package from where the thing we're registering is defined.

Reply:
all the registration now happened in their package `init` of the corresponding index file because:

- avoid importing cycle
- the `init` of other concreted packages you mentioned will not be executed because we use interface to call them and did not refer any concreted packages